cabouman/mbircone
# mace 3D & 4D PR

This is a pull request to merge mace3D and mace4D into master.

To test:
1. checkout pr/mace4D branch: `git checkout pr/mace4D`, `git pull`
2. Install requirements.txt: `pip install -r requirements.txt`
3. Re-install mbircone: `pip install .`
4. Install demo related packages: `cd demo`, `pip install -r requirements_demo.txt`
5. Clear system matrix cache: `rm -r ~/.cache/mbircone/`
6. Run demo:
- mace3D: python demo_mace3D_fast.py
- mace4D: python demo_mace4D_fast.py

Both demo should take about 5-10 minutes on cluster, and 15-20 minutes on a personal computer.


In addition, if you have access to the cluster and would like to perform a higher quality demo, you could run `demo_mace3D.py` and `demo_mace4D.py` (basically 3D/4D reconstructions with higher resolution phantoms).
